File: android/src/main/java/com/lightningkite/khrysalis/observables/binding/ProgressBar.binding.actual.kt
package com.lightningkite.khrysalis.observables.binding

import android.widget.ProgressBar
import com.lightningkite.khrysalis.observables.MutableObservableProperty
import com.lightningkite.khrysalis.observables.ObservableProperty
import com.lightningkite.khrysalis.observables.subscribeBy
import com.lightningkite.khrysalis.rx.removed
import com.lightningkite.khrysalis.rx.until

fun ProgressBar.bindInt(
    observable: ObservableProperty<Int>
){
    observable.subscribeBy { value ->
        this.progress = value
    }.until(this@bindInt.removed)
}
fun ProgressBar.bindFloat(
    observable: ObservableProperty<Float>
){
    observable.subscribeBy { value ->
        this.progress = (value.coerceIn(0.0f, 1.0f) * 100).toInt()
    }.until(this@bindFloat.removed)
}

File: ios/Khrysalis/src/observables/binding/ProgressBar.binding.actual.swift
//
//  UIActivityIndicatorView.binding.actual.swift
//  Khrysalis
//
//  Created by Brady on 8/11/20.
//

import Foundation
import UIKit

public extension UIProgressView{
    func bindInt(observable:ObservableProperty<Int>){
        observable.subscribeBy{value in
            self.progress = Float(value / 100)
        }.until(self.removed)
    }
    
    func bindLong(observable:ObservableProperty<Int64>){
        observable.subscribeBy{value in
            self.progress = Float(value / 100)
        }.until(self.removed)
    }
    
    
    func bindFloat(observable:ObservableProperty<Float>){
        observable.subscribeBy{value in
            if value > 1.0 {
                self.progress = 1.0
            } else if value < 0.0 {
                self.progress = 0.0
            }else{
                self.progress =  value
            }
        }.until(self.removed)
    }
}


